### PR TITLE
feat: adds initial cut at code for tutorial #3

### PR DESCRIPTION
**Pull request by Gowiem**
## what
* Moves to the centralized cloudposse/tutorials image pattern that we discussed
* Adds code for tutorial #3 

## why
* One centralized tutorials image will allow us to bundle all of our tutorials into one image
	* I can do the following if we like this idea: 
		- [ ] Update the other tutorials to follow this pattern
		- [ ] Update our documentation to explain this pattern
		- [ ] Add GH Action automation to start building + pushing the cloudposse/tutorials image on each merge to `main`
* Adds the code for tutorial #3: "Getting started with AWS + Atmos"
	* I'll start writing up the tutorial walk through now that I've walked through the code bit.

### CODE REVIEW

File: 03-first-aws-environment/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Tutorial #3: You first AWS Environment with Atmos
+
+The code in this directory is an accompaniment to the [TODO](TODO) tutorial in our SweetOps documentation.
```
Comment (osterman) [inline 03-first-aws-environment/README.md]: what will this link to?

File: 03-first-aws-environment/bin/uniquify.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
```
Comment (osterman) [inline 03-first-aws-environment/bin/uniquify.sh]: I understand why you have this script, but seems like we're doing something wrong if we need it. Maybe something we should do differently in the component.

File: 03-first-aws-environment/components/terraform/tfstate-backend/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+**Stack Level**: Regional
+
+Here's an example snippet for how to use this component.
+
+```yaml
+components:
+  terraform:
+    tfstate-backend:
+      vars:
+        name: tfstate
```
Comment (osterman) [inline 03-first-aws-environment/components/terraform/tfstate-backend/README.md:16]: we can also set attributes if we need a unique name.

File: rootfs/etc/motd [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
+
+=================================================
+Welcome to the SweetOps tutorials toolbox!
```
Comment (osterman) [inline rootfs/etc/motd:4]:
```suggestion
Welcome to the SweetOps tutorial toolbox!
```

File: rootfs/etc/motd [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+This Geodesic toolbox is an interactive way to work through the various tutorials from the SweetOps documentation available at https://docs.cloudposse.com/tutorials.
+It includes all of the code from our tutorials repository mounted to `/tutorials` which you can view at https://github.com/cloudposse/tutorials.
+Please open a GitHub issue if you run into any problems!
+=================================================
```
Comment (osterman) [inline rootfs/etc/motd]:
```suggestion
This Geodesic toolbox is an interactive way to work through the various tutorials from the SweetOps documentation available at https://docs.cloudposse.com/tutorials.

It includes all of the code from our tutorials repository mounted to `/tutorials` which you can view at https://github.com/cloudposse/tutorials.

Please open a GitHub issue if you run into any problems!
=================================================
```

File: 03-first-aws-environment/bin/uniquify.sh [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+# buckets to be unique, otherwise we'll run into issues with AWS S3 API errors blocking
+# our terraform applies. This scripts uniquifies all bucket names in the project.
+
+# Build random value
+RAND=$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 12 | head -n 1)
+PROD_RAND=$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 12 | head -n 1)
+
+# Updates our tfstate-backend's name to be unique
+yq eval --inplace \
+    ".components.terraform.tfstate-backend.vars.name = \"tfstate-$RAND\"" \
```
Comment (osterman) [inline 03-first-aws-environment/bin/uniquify.sh]: Maybe we set an attribute instead?

File: 03-first-aws-environment/bin/uniquify.sh (hunk 2 above)
Comment (Gowiem) [reply 03-first-aws-environment/bin/uniquify.sh]: @osterman Yeah, it's not the best situation. What I want is to be able to use an unchanged version of the `tfstate-backend` component, but unfortunately that produces consistent bucket names. If we want to introduce `random_pet` into both the `tfstate-backend` and `static-site` components to create unique names then we can do so, but then we'll require the tutorial followers to have to do manual updates to some of the stack files. Maybe that is fine? But it seemed undesirable. What would you like to do?

File: 03-first-aws-environment/components/terraform/tfstate-backend/README.md (hunk 3 above)
Comment (Gowiem) [reply 03-first-aws-environment/components/terraform/tfstate-backend/README.md:16]: Yeah good point. I can update, but let's go full circle on the conversation above regarding if we want to keep `uniquify` or not.

File: 03-first-aws-environment/bin/uniquify.sh (hunk 6 above)
Comment (Gowiem) [reply 03-first-aws-environment/bin/uniquify.sh]: Will do depending on outcome of our conversation above.

File: rootfs/etc/motd (hunk 4 above)
Comment (Gowiem) [reply rootfs/etc/motd:4]: @osterman I don't have the strongest opinion on this one, but if the image is going to be called `cloudposse/tutorials` then shouldn't we call it the tutorials toolbox? Maybe it does sound awkward though?

File: 03-first-aws-environment/README.md (hunk 1 above)
Comment (Gowiem) [reply 03-first-aws-environment/README.md]: I'll update with the link to https://github.com/cloudposse/docs/pull/537 when I do updates following our conversations below.

File: 03-first-aws-environment/bin/uniquify.sh (hunk 2 above)
Comment (osterman) [reply 03-first-aws-environment/bin/uniquify.sh]: What is the `tfstate-backend` writes the `stacks/catalog/backend.yaml` file with all the settings?

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+# Get version from https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst
+ARG AWS_CLI_VERSION=2.1.34
+RUN AWSTMPDIR=$(mktemp -d -t aws-inst-XXXXXXXXXX) && \
+    curl -sSL "https://awscli.amazonaws.com/awscli-exe-linux-x86_64-${AWS_CLI_VERSION}.zip" -o "$AWSTMPDIR/awscliv2.zip" && \
+    cd $AWSTMPDIR && \
+    unzip -qq awscliv2.zip && \
+    ./aws/install -i /usr/share/aws/v2 -b /usr/share/aws/v2/bin && \
+    update-alternatives --install /usr/local/bin/aws aws /usr/share/aws/v2/bin/aws 2 && \
+    rm -rf $AWSTMPDIR
+
```
Comment (osterman) [inline Dockerfile:21]:
```suggestion

# Install ubuntu universe repo so we can install more helpful packages
RUN apt-get install -y software-properties-common && \
    add-apt-repository "deb http://archive.ubuntu.com/ubuntu bionic universe" && \
    gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv 3B4FE6ACC0B21F32 && \
    gpg --export --armor 3B4FE6ACC0B21F32 | apt-key add - && \
    apt-get update && \
    apt-get install -y golang-petname


```

File: 03-first-aws-environment/bin/uniquify.sh [hunk 8]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+#
+# Since we're creating an S3 bucket as part of this tutorial, we need the names of those
+# buckets to be unique, otherwise we'll run into issues with AWS S3 API errors blocking
+# our terraform applies. This scripts uniquifies all bucket names in the project.
+
+# Build random value
+RAND=$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 12 | head -n 1)
+PROD_RAND=$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 12 | head -n 1)
```
Comment (osterman) [inline 03-first-aws-environment/bin/uniquify.sh]:
```suggestion
RAND=$(golang-petname)
PROD_RAND=$(golang-petname)
```

File: rootfs/etc/motd (hunk 4 above)
Comment (osterman) [reply rootfs/etc/motd:4]: Ya, same no strong feelings. We can leave it.

File: 03-first-aws-environment/bin/uniquify.sh [hunk 9]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+#
+# Since we're creating an S3 bucket as part of this tutorial, we need the names of those
+# buckets to be unique, otherwise we'll run into issues with AWS S3 API errors blocking
+# our terraform applies. This scripts uniquifies all bucket names in the project.
+
+# Build random value
+RAND=$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 12 | head -n 1)
```
Comment (osterman) [inline 03-first-aws-environment/bin/uniquify.sh]:
```suggestion
RANDOM_PET=$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 12 | head -n 1)
```

File: 03-first-aws-environment/bin/uniquify.sh [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    ".terraform.backend.s3.dynamodb_table = \"acme-uw2-tfstate-$RAND-lock\"" \
+    stacks/catalog/globals.yaml
+
+# Updates our dev + prod static-site component vars to ensure we're creating unique names
+yq eval --inplace \
+    ".components.terraform.static-site.vars.attributes.[0] = \"$RAND\"" \
+    stacks/uw2-dev.yaml
+
+yq eval --inplace \
+    ".components.terraform.static-site.vars.attributes.[0] = \"$PROD_RAND\"" \
```
Comment (osterman) [inline 03-first-aws-environment/bin/uniquify.sh]: We shouldn't need this since prod would be disambiguated by the stage name differing.

Comment (osterman) [reply 03-first-aws-environment/bin/uniquify.sh]: Hoping we can only have one random pet =)

File: 03-first-aws-environment/bin/uniquify.sh (hunk 2 above)
Comment (osterman) [reply 03-first-aws-environment/bin/uniquify.sh]: I'm okay with this script approach. Let's ship this sooner. 

Can we just rename this script to `random-pet.sh`. The `uniquify` script doesn't have such a nice ring to it =)

File: 03-first-aws-environment/bin/uniquify.sh (hunk 10 above)
Comment (Gowiem) [reply 03-first-aws-environment/bin/uniquify.sh]: Ah solid point 👍 Will update.

**Review by osterman — APPROVED** [review]

**Pull request merged**